src/cascadia/TerminalApp/TerminalPage.h

```diff
@@ -95,8 +95,12 @@ namespace winrt::TerminalApp::implementation
         std::deque<winrt::TerminalApp::ActionAndArgs> _startupActions;
         winrt::fire_and_forget _ProcessStartupActions();
 
+        std::function<void()> _acceptPaste;
```

Comment:
I don't know what you think about storing a `std::function` here instead of storing the text and the callback and calling the callback with the text in the `TerminalPage::_AcceptPasteButtonOnClick` handler.

I did it this way because if we stored the text and the callback, it would looks more confusing IMO:
```c++
winrt::hstring _acceptPasteText;
winrt::Microsoft::Terminal::TerminalControl::PasteFromClipboardEventArgs _acceptPasteArgs;
```

Reply:
It's clever, but per above comment if we make dialog presentation waitable we can keep the paste acceptance inline